internal/resources/cluster/resource_cluster.go

```diff
@@ -161,6 +163,12 @@ func constructSpec(d *schema.ResourceData) (spec *clustermodel.VmwareTanzuManage
 		spec.ProxyName = v.(string)
 	}
 
```

Comment:
Should we throw an error if none of the TKG-AWS, TKGS-vSphere or TKG-vSphere specs is configured?

Reply:
Terraform takes care of the error handling when we do not populate any of the specs as those are required fields. The following error is seen: `POST request failed with status : 400 Bad Request, response: {"error":"Invalid value for management cluster name and/or provisioner name in request","code":3`